src/main/java/put/apl/Algorithms/Sorting/Implementation/BucketSort.java

```diff
@@ -0,0 +1,55 @@
+package put.apl.Algorithms.Sorting.Implementation;
+
+import org.springframework.stereotype.Component;
+import put.apl.Algorithms.Sorting.Data.SortingData;
+import put.apl.Algorithms.Sorting.SortingResult;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Component("bucketSort")
+public class BucketSort implements SortingAlgorithm {
+
+    @Override
+    public SortingResult sort(SortingData tab) {
+        int max = Integer.MIN_VALUE;
```

Comment:
Jeśli nie używasz swapów i compów taba, to powinna algorytm powinien być w stanie wyrzucić InterruptException sprawdzając często w każdej pętli czy wątek is interrupted. W sumie najlepiej by było funkcję to tego zaimplementować w klasie SortingAlgorithm i ją wywoływać przy okazji każdej pęrli